mobile/src/lib.rs

```diff
@@ -12,14 +12,18 @@ fn setup_logging() {
     use tracing::level_filters::LevelFilter;
     use tracing_subscriber::filter::Targets;
     use tracing_subscriber::layer::SubscriberExt;
-    use tracing_subscriber::util::SubscriberInitExt;
     let targets = Targets::new()
         .with_default(LevelFilter::INFO)
         .with_target("mycelium::router", LevelFilter::WARN);
-    tracing_subscriber::registry()
+    let subscriber = tracing_subscriber::registry()
         .with(tracing_android::layer("mycelium").expect("failed to setup logger"))
-        .with(targets)
-        .init();
+        .with(targets);
+    if let Err(e) = tracing::subscriber::set_global_default(subscriber) {
+        // the error is expected to happen on second start, we can ignore it.
+        if e.to_string() != "a global default trace dispatcher has already been set" {
+            error!("Failed to set global default subscriber: {}", e);
+        }
```

Comment:
I'm not really a fan of checking based on error string equivalence. Considering this is the only thing which can go wrong, I'd prefer to just check if the call result is an error (`set_global_default(...).is_err()`) instead.

Reply:
> I'm not really a fan of checking based on error string equivalence

actually, me as well

> I'd prefer to just check if the call result is an error (set_global_default(...).is_err()) instead.

i think it is basically the same with what i currently do, no?

I will take a look at `once` or something similar

Reply:
oh, i think i misunderstood you

> Considering this is the only thing which can go wrong

I'm not sure about this actually, that is why i'm checking for specific error.

Reply:
i improve it using lazy_static, pls check again @LeeSmet